### PR TITLE
fix(utils): make `looseEqual()` util compliant with Vue.js spec

### LINKED ISSUE #2682: SSR fails with checkbox @ 2.0.0-rc.13
I'm using nuxt, page with `b-checkbox` on it fails with `Error in render: "ReferenceError: File is not defined"` error.

https://github.com/bootstrap-vue/bootstrap-vue/commit/401d3e974249c404df2d80282ccecae71baf57a7
I suppose this is the culprit, File isn't available in node.js environment.

### PR DESCRIPTION
**Pull request by jackmu95**
## Description of Pull Request:

This PR makes the `looseEqual()` compliant to the Vue.js spec and improves performance slightly.

Ref: https://github.com/vuejs/vue/pull/9527

Closes #2682.

## PR checklist:

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Enhancement to an existing feature
- [ ] ARIA accessibility
- [ ] Documentation update
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch, **not** the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (i.e.
      `fixes #xxxx[,#xxxx]`, where "xxxx" is the issue number)
- [x] The PR should address only one issue or feature. If adding multiple features or fixing a bug
      and adding a new feature, break them into separate PRs if at all possible.
- [x] PR titles should following the
      [**Conventional Commits**](https://www.conventionalcommits.org/) naming convention (i.e.
      "fix(alert): not alerting during SSR render", "docs(badge): Updated pill examples, fix typos",
      "chore: fix typo in docs", etc). **This is very important, as the `CHANGELOG` is generated
      from these messages.**

**If new features/enhancement/fixes are added or changed:**

- [ ] Includes documentation updates (including updating the component's `package.json` for slot and
      event changes)
- [ ] New/updated tests are included and passing (if required)
- [ ] Existing test suites are passing
- [ ] The changes have not impacted the functionality of other components or directives
- [ ] ARIA Accessibility has been taken into consideration (does it affect screen reader users or
      keyboard only users? clickable items should be in the tab index, etc)

**If adding a new feature, or changing the functionality of an existing feature, the PR's
description above includes:**

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a
      suggestion issue first and wait for approval before working on it)

### CODE REVIEW

**Comment by jackmu95** [comment]
@tmorehouse I think we can merge this independent from the Vue.js PR because it's compliant with their spec and fixes #2682.

File: src/utils/loose-equal.spec.js [hunk 1]
```diff
@@ -61,14 +61,14 @@ describe('looseEqual', async () => {
     // Identical file object references
     expect(looseEqual(file1, file1)).toBe(true)
     // Different file references with identical values
-    expect(looseEqual(file1, file2)).toBe(false)
+    expect(looseEqual(file1, file2)).toBe(true)
```
[the remark this answers is not in the document]
Comment (tmorehouse) [reply src/utils/loose-equal.spec.js]: I didn't do a test to see if two identically named files with same properties/size, but different content would trigger an update to the v-model of form-file.

[the remark this answers is not in the document]
Comment (jackmu95) [reply src/utils/loose-equal.spec.js]: Will look at this a it further.

[the remark this answers is not in the document]
Comment (tmorehouse) [reply src/utils/loose-equal.spec.js]: Just added a test for it...

[the remark this answers is not in the document]
Comment (tmorehouse) [reply src/utils/loose-equal.spec.js]: And the test is failing... :disappointed:

[the remark this answers is not in the document]
Comment (jackmu95) [reply src/utils/loose-equal.spec.js]: And it failed...

[the remark this answers is not in the document]
Comment (tmorehouse) [reply src/utils/loose-equal.spec.js]: Maybe the other solution is to remove the `looseEqual` check in `<b-form-file>` 
 watcher: https://github.com/bootstrap-vue/bootstrap-vue/blob/9e824a510d0b8a3022a4e926f870e914d7072621/src/components/form-file/form-file.js#L87-L96

[the remark this answers is not in the document]
Comment (tmorehouse) [reply src/utils/loose-equal.spec.js]: i.e. remove lines 88-90 in `form-file.js`

[the remark this answers is not in the document]
Comment (jackmu95) [reply src/utils/loose-equal.spec.js]: We can't do that. See #2289.

[the remark this answers is not in the document]
Comment (tmorehouse) [reply src/utils/loose-equal.spec.js]: Ah yeah...

Hmmm... what about if we do the following check in form-file.js:

```js
if (isArray(newVal) && isArray(oldVal) && newVal.length === 0 && oldVal.length === 0) {
  return
}
```

[the remark this answers is not in the document]
Comment (jackmu95) [reply src/utils/loose-equal.spec.js]: We need a check for equal files too.
```js
if (
  newVal === oldVal ||
  (isArray(newVal) && isArray(oldVal) && newVal.length === 0 && oldVal.length === 0)
) {
  return
}
´´´

[the remark this answers is not in the document]
Comment (jackmu95) [reply src/utils/loose-equal.spec.js]: Looked at the `looseEqual()` File handling a bit further and even the `size` of the File gets compared. The only two edge cases would be same file from different paths and files with equal props which size is the same.

[the remark this answers is not in the document]
Comment (tmorehouse) [reply src/utils/loose-equal.spec.js]: Yeah... which could bite some users by surprise.

Thinking we need to maybe re-think the watcher in form-file, maybe by:

```js
if (newVal === oldVal) {
  return
}
if (
  isArray(newVal) &&
  isArray(oldVal) &&
  newVal.length === oldVal.length &&
  newVal.every((v, i) => v === oldVal[i])
) {
 return
}
// ...
```

We need to check the lengths for equality before running the `newVal.every(...)`

[the remark this answers is not in the document]
Comment (tmorehouse) [reply src/utils/loose-equal.spec.js]: PR #2695 fixes `b-form-file`.

So we could probably remove the tests for `File` in `loose-equal.spec.js`

**Pull request merged**